File: go/sum-of-multiples/sum_of_multiples.go
package summultiples

const testVersion = 2

func SumMultiples(limit int, divisors ...int) (sum int) {
	for i := 0; i < limit; i++ {
		for _, d := range divisors {
			if i%d == 0 {
				sum += i
				break
			}
		}
	}
	return sum
}

File: haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = sum $ filter isMultipleOfAnyFactor [1..(limit-1)]
    where isMultipleOfAnyFactor x = any (divisible x) factors
          divisible x y = x `mod` y == 0 
